paddle/understand_sentiment_lstm.py

```diff
@@ -0,0 +1,172 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import numpy as np
+import argparse
+import time
+
+import paddle.v2 as paddle
+import paddle.v2.fluid as fluid
+import paddle.v2.fluid.profiler as profiler
+
+
+def parse_args():
+    parser = argparse.ArgumentParser("LSTM model benchmark.")
+    parser.add_argument(
+        '--batch_size', type=int, default=32, help='The minibatch size.')
+    parser.add_argument(
+        '--stacked_num', type=int, default=1, help='Stacked LSTM Layer num.')
+    parser.add_argument(
+        '--emb_dim', type=int, default=32, help='The embedding dim.')
+    parser.add_argument(
+        '--seq_len',
+        type=int,
+        default=80,
+        help='The sequence length of one sentence.')
+    parser.add_argument(
+        '--iterations', type=int, default=35, help='The number of minibatches.')
+    parser.add_argument(
+        '--pass_num', type=int, default=100, help='The number of passes.')
+    parser.add_argument(
+        '--device',
+        type=str,
+        default='GPU',
+        choices=['CPU', 'GPU'],
+        help='The device type.')
+    parser.add_argument(
+        '--infer_only', action='store_true', help='If set, run forward only.')
+    parser.add_argument(
+        '--use_cprof', action='store_true', help='If set, use cProfile.')
+    parser.add_argument(
+        '--use_nvprof',
+        action='store_false',
+        help='If set, use nvprof for CUDA.')
+    args = parser.parse_args()
+    return args
+
+
+def print_arguments(args):
+    print('-----------  Configuration Arguments -----------')
+    for arg, value in sorted(vars(args).iteritems()):
+        print('%s: %s' % (arg, value))
+    print('------------------------------------------------')
+
+
+def lstm_model(data, dict_dim, class_dim=2):
+    batch_size = args.batch_size
+    emb_dim = args.emb_dim
+    seq_len = args.seq_len
+    stacked_num = args.stacked_num
+
+    emb = fluid.layers.embedding(input=data, size=[dict_dim, emb_dim])
+    emb = fluid.layers.reshape(x=emb, shape=[batch_size, seq_len, args.emb_dim])
+    emb = fluid.layers.transpose(x=emb, axis=[1, 0, 2])
+
+    c_pre_init = fluid.layers.fill_constant(
+        dtype=emb.dtype, shape=[batch_size, emb_dim], value=0.0)
+    layer_1_out = emb
+    for i in range(stacked_num):
+        layer_1_out = fluid.layers.lstm(
+            layer_1_out, c_pre_init=c_pre_init, hidden_dim=emb_dim)
+        layer_1_out = fluid.layers.transpose(x=layer_1_out, axis=[1, 0, 2])
```

Comment:
The `fluid.layers.transpose` should be in the outside of for loop.

Reply:
fixed.